File: blog-backend/app/content/service/internal/service/category.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-blog/gen/api/go/common/pagination"
	v1 "kratos-blog/gen/api/go/content/service/v1"

	"kratos-blog/app/content/service/internal/biz"
)

type CategoryService struct {
	v1.UnimplementedCategoryServiceServer

	uc  *biz.CategoryUseCase
	log *log.Helper
}

func NewCategoryService(logger log.Logger, uc *biz.CategoryUseCase) *CategoryService {
	l := log.NewHelper(log.With(logger, "module", "category/service/content-service"))
	return &CategoryService{
		log: l,
		uc:  uc,
	}
}

// ListCategory 获取类别列表
func (s *CategoryService) ListCategory(ctx context.Context, req *pagination.PagingRequest) (*v1.ListCategoryResponse, error) {
	return s.uc.List(ctx, req)
}

// GetCategory 获取类别数据
func (s *CategoryService) GetCategory(ctx context.Context, req *v1.GetCategoryRequest) (*v1.Category, error) {
	return s.uc.Get(ctx, req)
}

// CreateCategory 创建类别
func (s *CategoryService) CreateCategory(ctx context.Context, req *v1.CreateCategoryRequest) (*v1.Category, error) {
	return s.uc.Create(ctx, req)
}

// UpdateCategory 更新类别
func (s *CategoryService) UpdateCategory(ctx context.Context, req *v1.UpdateCategoryRequest) (*v1.Category, error) {
	return s.uc.Update(ctx, req)
}

// DeleteCategory 删除类别
func (s *CategoryService) DeleteCategory(ctx context.Context, req *v1.DeleteCategoryRequest) (*emptypb.Empty, error) {
	_, err := s.uc.Delete(ctx, req)
	if err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}

File: frontend/admin/src/api/blog/category.ts
import { defHttp } from '/@/utils/http/axios';

const route = {
  Categories: '/categories',
  CategoryWithId: (id) => `/categories/${id}`,
};

// 获取列表
export const ListCategory: ListCategory = (params) => {
  return defHttp.get<ListCategoryResponse>({
    url: route.Categories,
    params,
  });
};

// 获取
export const GetCategory: GetCategory = (params) => {
  return defHttp.get<Category>({
    url: route.CategoryWithId(params.id),
  });
};

// 创建
export const CreateCategory: CreateCategory = (params) => {
  return defHttp.post<Category>(
    { url: route.Categories, params },
    {
      errorMessageMode: 'none',
    },
  );
};

// 更新
export const UpdateCategory: UpdateCategory = (params) => {
  return defHttp.put<Category>({
    url: route.CategoryWithId(params.id),
    data: params.category,
  });
};

// 删除
export const DeleteCategory: DeleteCategory = (params) => {
  return defHttp.delete({
    url: route.CategoryWithId(params.id),
  });
};
